cli/command_repository_upgrade.go

```diff
@@ -269,6 +269,12 @@ func (c *commandRepositoryUpgrade) upgrade(ctx context.Context, rep repo.DirectR
 	if err := rep.SetParameters(ctx, mp, rep.BlobCfg()); err != nil {
 		return errors.Wrap(err, "error setting parameters")
 	}
+
+	// poison V0 index so that old readers won't be able to open it.
+	if err := content.WriteLegacyIndexPoisonBlob(ctx, rep.BlobStorage()); err != nil {
```

Comment:
@jkowalski planning to write compat tests to hit this and verify with older binaries ?

Reply:
looks like repo-tests have some coverage, LGTM